provider/provider.go

```diff
@@ -59,6 +61,23 @@ func StartProvider(flags *pflag.FlagSet) {
 	logger.SetupLogging(logLevel)
 	logger.ProviderLogger.LogInfo("provider_setup", fmt.Sprintf("Starting provider on port `%v`", config.ProviderConfig.Port))
 
+	// Check if the default workspace exists
+	defaultWorkspace, err := db.GetDefaultWorkspace()
```

Comment:
I don't think there is a point to do this for the provider, you can't even start a provider instance without starting a hub instance to create the config :P

Reply:
In a scenario where the user starts the provider before the hub and the default workspace doesn't exist, the provider needs to create the default workspace. Additionally, the provider should assign its devices to the default workspace if they have no existing assignment.

Reply:
Yeah, that makes sense but you are required to have a provider configuration added to the hub to be able to start it, if you don't provide `--nickname` flag or its invalid, then the provider instance will not boot up at all? :)

Reply:
Right, but I’m thinking of a scenario where the user is running GADS from previous versions and already has MongoDB configured with all the data required for GADS to run

Reply:
If we agree that only the hub should create the default workspace, as we can have many providers and potential issues in the future regarding workspace creation on the provider side, I can remove this feature from the provider.

Reply:
You are right, it is better to tighten our pants with this. At some point after multiple releases when this should no longer be a case, we might remove it but let's leave it for now 👌